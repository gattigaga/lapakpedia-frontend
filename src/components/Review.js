import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { format } from "date-fns";

import Avatar from "./Avatar";
import Rating from "./Rating";

const Container = styled.div`
  display: flex;
  border: 1px solid #ddd;
  padding: 16px;
  margin-bottom: 24px;
`;

const Content = styled.div`
  margin-left: 18px;
`;

const Timestamp = styled.p`
  font-family: Roboto;
  font-size: 11px;
  color: #aaa;
  margin-top: 4px;
  margin-bottom: 0px;
`;

const Text = styled.p`
  font-family: Roboto;
  font-size: 14px;
  margin-top: 16px;
  margin-bottom: 0px;
`;

const Review = ({ rating, time, text, reviewer, avatar }) => (
  <Container>
    <Avatar src={avatar} />
    <Content>
      <Rating value={rating} isDisabled />
      <Timestamp>
        by {reviewer}, {format(time, "DD MMMM YYYY, HH.mm a")}
      </Timestamp>
      <Text>{text}</Text>
    </Content>
  </Container>
);

Review.propTypes = {
  rating: PropTypes.number,
  time: PropTypes.string,
  text: PropTypes.string,
  reviewer: PropTypes.string,
  avatar: PropTypes.string
};

export default Review;
