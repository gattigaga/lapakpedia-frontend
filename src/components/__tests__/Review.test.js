import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

import Review from "components/Review";

describe("Review", () => {
  jest.mock("../Avatar", () => "Avatar");
  jest.mock("../Rating", () => "Rating");

  const setup = propOverrides => {
    const props = {
      rating: 4,
      time: "2018-05-06T15:00:14.376Z",
      text: "I love this product",
      reviewer: "markzuckerberg",
      avatar: require("assets/images/dummy240x240.png"),
      ...propOverrides
    };

    const wrapper = shallow(<Review {...props} />);

    return {
      props,
      wrapper
    };
  };

  it("should renders default", () => {
    const { wrapper } = setup();

    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
