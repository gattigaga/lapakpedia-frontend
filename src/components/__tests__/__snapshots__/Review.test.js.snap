// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Review should renders default 1`] = `
<styled.div>
  <Avatar
    alt="Profile"
    size={48}
    src="dummy240x240.png"
  />
  <styled.div>
    <Rating
      isDisabled={true}
      value={4}
    />
    <styled.p>
      by 
      markzuckerberg
      , 
      06 May 2018, 22.00 pm
    </styled.p>
    <styled.p>
      I love this product
    </styled.p>
  </styled.div>
</styled.div>
`;
